spec/opsgenie_alert_tagging_spec.rb

```diff
@@ -0,0 +1,138 @@
+require 'spec_helper'
+require './lib/opsgenie_alert_tagging'
+
+RSpec.describe OpsgenieAlertTagging do
+  before(:each) do
+    allow(HTTParty).to receive(:get).and_return("data" => fake_opsgenie_alerts_empty_tags)
+  end
+
+  let(:subject) { OpsgenieAlertTagging.new("2021-9-2") }
+
+  describe '#list_all_alerts' do
+    it 'returns alerts from OpsGenie for a given date' do
+      expect(HTTParty).to receive(:get)
+        .with(
+          "https://api.opsgenie.com/v2/alerts?query=createdAt%3A22-08-2021&sort=createdAt&order=desc",
+          headers: {
+            "Content-Type" => "application/json",
+            "Authorization" => anything
+          }
+        )
+        .and_return("data" => fake_opsgenie_alerts_empty_tags)
+
+      result = subject.list_all_alerts("2021-08-22")
+
+      expect(result).to eq(fake_opsgenie_alerts_empty_tags)
+    end
+  end
+
+  
+  def fake_opsgenie_alerts_empty_tags
+    [
+      fake_alert_inhours_weekday,
+      fake_alert_wakinghours_weekday,
+      fake_alert_sleepinghours,
+      fake_alert_wakinghours_weekend
+    ]
+  end
+
+  def fake_alert_inhours_weekday
+    {"seen"=>true,
+      "id"=>"ccd0bcac-2235-4d9f-a043-5f8ad111cf45-1629609505865",
+      "tinyId"=>"3255",
+      "alias"=>"ccd0bcac-2235-4d9f-a043-5f8ad111cf45-1629609505865",
+      "message"=>"[Updown.io] Timeout was reached",
+      "status"=>"closed",
+      "acknowledged"=>false,
+      "isSeen"=>true,
+      "tags"=>[],
+      "snoozed"=>false,
+      "count"=>1,
+      "lastOccurredAt"=>"2021-08-23T11:18:25.865Z",
+      "createdAt"=>"2021-08-23T11:18:25.865Z",
+      "updatedAt"=>"2021-09-02T12:08:50.83Z",
+      "source"=>"UpdownIO",
+      "owner"=>"",
+      "priority"=>"P3",
+      "teams"=>[{"id"=>"72c99d92-c7c8-4fd9-b8db-9cc293cd19e7"}],
+      "responders"=>[{"type"=>"team", "id"=>"72c99d92-c7c8-4fd9-b8db-9cc293cd19e7"}],
+      "integration"=>{"id"=>"ae438e99-b745-452d-a8a3-14b766e0c3fd", "name"=>"Working Hours - updown.io", "type"=>"UpdownIO"},
+      "report"=>{"ackTime"=>274757772, "closeTime"=>274757772, "closedBy"=>"bob@dxw.com"},
+      "ownerTeamId"=>"72c99d92-c7c8-4fd9-b8db-9cc293cd19e7"}
+  end
```

Comment:
Can we have simpler test cases? We don't care about most of these properties, and it's very hard to see what makes this one different from the next one.